### PR TITLE
Updated for Mediation 5

### PR DESCRIPTION
**Pull request by bwised**
This PR updates the Kotlin/compose demo for Chartboost Mediation 5 compatibility including initializing the SDK with Core.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,24 +17,36 @@ To quickly get started, simply add the following dependencies inside your app's
...
 dependencies {
     ...
 
-    implementation 'com.chartboost:chartboost-mediation-sdk:4.9.0'
+	// Chartboost Mediation SDK
+    implementation("com.chartboost:chartboost-mediation-sdk:5.+")
+    implementation("com.chartboost:chartboost-core-sdk:1.+")
+	
+	// Chartboost Mediation Adapters
+    implementation("com.chartboost:chartboost-mediation-adapter-chartboost:5.+")
```
Comment (cb-jpadilla) [inline README.md:38]:
```suggestion
    // Chartboost Mediation SDK
    implementation("com.chartboost:chartboost-mediation-sdk:5.+")
    implementation("com.chartboost:chartboost-core-sdk:1.+")
	
    // Chartboost Mediation Adapters
    implementation("com.chartboost:chartboost-mediation-adapter-chartboost:5.+")
```

File: README.md [hunk 2]
```diff
@@ -65,261 +77,140 @@ implementation "com.google.android.gms:play-services-ads-identifier:18.0.1"
...
-    new HeliumInitializationOptions(),
-    new HeliumSdk.HeliumSdkListener() {
-        @Override
-        public void didInitialize(Error error) {
-            if (error != null) {
-                Log.d(TAG,"Chartboost Mediation SDK failed to initialize. Reason: " + error.getMessage());
-            } else {
-                //SDK Started,
-                Log.d(TAG,"Chartboost Mediation SDK initialized successfully");
+```kotlin Kotlin
```
Comment (cb-jpadilla) [inline README.md:80]: If we are updating the code in the README.md, we may need to update the text as well as it is referencing old names.

File: compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt [hunk 3]
```diff
@@ -554,7 +321,7 @@ object AdController : DefaultLifecycleObserver {
             }
 
             logState.add("Fullscreen ad is about to show")
-            val result = fullscreenAd?.show(context)
+            val result = fullscreenAd?.show(context as Activity)
```
Comment (chauduyphanvu) [inline compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt:324]: In Kotlin this is not a safe cast. https://kotlinlang.org/docs/typecasts.html#safe-nullable-cast-operator

File: compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt [hunk 4]
```diff
@@ -161,15 +150,13 @@ object AdController : DefaultLifecycleObserver {
...
-     * @param shouldUseFullscreenApi True if the fullscreen API should be used, false otherwise.
      * @param showBtnEnabled True if the corresponding show button should be enabled, false otherwise.
      */
     fun showAd(
         context: Context,
         adType: AdType,
         placementName: String,
         logState: MutableList<String>,
-        shouldUseFullscreenApi: MutableState<Boolean>,
         shouldUseFullscreenAdQueue: MutableState<Boolean>,
```
Comment (chauduyphanvu) [inline compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt:160]: I know this was not in scope but `shouldUseFullscreenAdQueue` is missing docs.

File: compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt [hunk 5]
```diff
@@ -120,15 +113,13 @@ object AdController : DefaultLifecycleObserver {
...
-     * @param shouldUseFullscreenApi True if the fullscreen API should be used, false otherwise.
      * @param showBtnEnabled True if the corresponding show button should be enabled, false otherwise.
      */
     fun loadAd(
         context: Context,
         adType: AdType,
         placementName: String,
         logState: MutableList<String>,
-        shouldUseFullscreenApi: MutableState<Boolean>,
         shouldUseFullscreenAdQueue: MutableState<Boolean>,
```
Comment (chauduyphanvu) [inline compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt:123]: I know this was not in scope but `shouldUseFullscreenAdQueue` is missing docs.

File: compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt [hunk 6]
```diff
@@ -85,32 +75,35 @@ object AdController : DefaultLifecycleObserver {
...
      * @param logState The log state to update.
      *
      * @return A result indicating whether the SDK was initialized successfully.
      */
     suspend fun initialize(
         context: Context,
         appId: String,
-        appSignature: String,
         logState: MutableList<String>,
-    ): Result<Unit> {
```
Comment (chauduyphanvu) [inline compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt:98]: This function no longer returns.

**Review by chauduyphanvu — APPROVED** [review]
some nits. lgtm otherwise.

File: compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt (hunk 5 above)
Comment (bwised) [reply compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt:122]: https://github.com/ChartBoost/chartboost-mediation-android-sdk-demo/pull/39/commits/8ff6389acd4d2fd66016cd2365bc5d6131463bf1

File: compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt (hunk 4 above)
Comment (bwised) [reply compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt:160]: https://github.com/ChartBoost/chartboost-mediation-android-sdk-demo/pull/39/commits/8ff6389acd4d2fd66016cd2365bc5d6131463bf1

File: compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt (hunk 3 above)
Comment (bwised) [reply compose/ChartboostMediationDemoCompose/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt]: https://github.com/ChartBoost/chartboost-mediation-android-sdk-demo/pull/39/commits/fbde09ae84f9524f054bdb4485c85ead12cf66d6

File: README.md (hunk 2 above)
Comment (bwised) [reply README.md:80]: https://github.com/ChartBoost/chartboost-mediation-android-sdk-demo/pull/39/commits/6c2d00d5572474dfef9c895c48988c429f192777

**Review by chauduyphanvu — APPROVED** [review]

**Pull request merged**